consensus/istanbul/core/roundstate_db.go

```diff
@@ -146,19 +191,28 @@ func (rsdb *roundStateDBImpl) GetLastView() (*istanbul.View, error) {
 		return nil, err
 	}
 
-	var entry istanbul.View
-	if err = rlp.DecodeBytes(rawEntry, &entry); err != nil {
-		return nil, err
-	}
-	return &entry, nil
+	return key2View(rawEntry), nil
 }
 
-func (rsdb *roundStateDBImpl) GetRoundStateFor(view *istanbul.View) (RoundState, error) {
-	viewKey, err := rlp.EncodeToBytes(view)
-	if err != nil {
+func (rsdb *roundStateDBImpl) GetOldestValidView() (*istanbul.View, error) {
+	lastView, err := rsdb.GetLastView()
+	// If nothing stored all views are valid
+	if err == leveldb.ErrNotFound {
+		return &istanbul.View{Sequence: common.Big0, Round: common.Big0}, nil
+	} else if err != nil {
 		return nil, err
 	}
 
+	oldestValidSequence := new(big.Int).Sub(lastView.Sequence, new(big.Int).SetUint64(rsdb.opts.sequencesToSave))
+	if oldestValidSequence.Cmp(common.Big0) < 0 {
+		oldestValidSequence = common.Big0
+	}
+
+	return &istanbul.View{Sequence: oldestValidSequence, Round: common.Big0}, nil
```

Comment:
is the round as 0 intentional?

Reply:
Yes, basically I'm want to say that all round state beginning on seq = `oldestValidSequence` are valid. So, round `0` and above